src/Renderer/DrawBlock/SheetUpdateHelpers.fs

```diff
@@ -245,12 +244,15 @@ let appendUndoList (undoList: Model List) (model_in: Model): Model List =
     let rec removeLast lst =
         match lst with
         | _ :: lst when List.isEmpty lst -> []
-        | hd :: lst -> hd :: (removeLast lst)
+        | hd :: tl -> hd :: (removeLast tl)
         | [] -> []
```

Comment:
Use list functions. not custom recursive function `List.truncate `will work, and never fail if length is `(max 0 (lst.Length - 1))`